dev/tools/examples_smoke_test.dart

```diff
@@ -0,0 +1,254 @@
+    stderr.writeln('Example smoke tests are only designed to run on desktop platforms');
+    exitCode = 4;
+    return;
+  }
+  final Directory flutterDir = filesystem.directory(
+    path.absolute(
+      path.dirname(
+        path.dirname(
+          path.dirname(platform.script.toFilePath()),
+        ),
+      ),
+    ),
+  );
+  final Directory apiDir = flutterDir.childDirectory('examples').childDirectory('api');
+  final File integrationTest = await generateTest(apiDir);
+  try {
+    await runSmokeTests(flutterDir: flutterDir, integrationTest: integrationTest, apiDir: apiDir);
+  } finally {
+    await cleanUp(integrationTest);
+  }
+}
+
+Future<void> cleanUp(File integrationTest) async {
+  try {
+    await integrationTest.delete();
+    // Delete the integration_test directory if it is empty.
+    await integrationTest.parent.delete(recursive: false);
+  } on FileSystemException {
+    // Ignore, there might be other files in there preventing it from
+    // being removed, or it might not exist.
+  }
+}
+
+// Executes the generated smoke test.
+Future<void> runSmokeTests({
+  required Directory flutterDir,
+  required File integrationTest,
+  required Directory apiDir,
+}) async {
+  final File flutterExe =
+      flutterDir.childDirectory('bin').childFile(platform.isWindows ? 'flutter.bat' : 'flutter');
+  final List<String> cmd = <String>[
+    // If we're in a container with no X display, then use the virtual framebuffer.
+    if (platform.isLinux &&
+        (platform.environment['DISPLAY'] == null ||
+         platform.environment['DISPLAY']!.isEmpty)) '/usr/bin/xvfb-run',
+    flutterExe.absolute.path,
+    'test',
+    '--reporter=expanded',
+    '--timeout=20m',
+    '--device-id=${platform.operatingSystem}',
+    integrationTest.absolute.path,
+  ];
+  await runCommand(cmd, workingDirectory: apiDir);
+}
+
+// A class to hold information related to an example, used to generate names
+// from for the tests.
+class ExampleInfo {
+  ExampleInfo(this.file, Directory examplesLibDir)
+      : importPath = _getImportPath(file, examplesLibDir),
+        importName = '' {
+    importName = importPath.replaceAll(RegExp(r'\.dart$'), '').replaceAll(RegExp(r'\W'), '_');
+  }
+
+  final File file;
+  final String importPath;
+  String importName;
+
+  static String _getImportPath(File example, Directory examplesLibDir) {
+    final String relativePath =
+        path.relative(example.absolute.path, from: examplesLibDir.absolute.path);
+    // So that Windows paths are proper URIs in the import statements.
+    return path.toUri(relativePath).toFilePath(windows: false);
+  }
+}
+
+// Generates the combined smoke test.
+Future<File> generateTest(Directory apiDir) async {
+  final Directory examplesLibDir = apiDir.childDirectory('lib');
+
+  // Get files from git, to avoid any non-repo files that might be in someone's
+  // workspace.
+  final List<String> gitFiles = (await runCommand(
+    <String>['git', 'ls-files', '**/*.dart'],
```

Comment:
does the glob and path separator work on windows?

Reply:
Yes, in git it does.

Reply:
ahh, so the glob is expanded by git?

Reply:
Yep.